ext/ImplicitDifferentiationChainRulesExt.jl

```diff
@@ -23,11 +23,11 @@ function ChainRulesCore.rrule(
     n, m = length(x), length(y)
 
     backend = ReverseRuleConfigBackend(rc)
-    pbA = pullback_function(backend, _y -> conditions(x, _y, z; kwargs...), y)
-    pbB = pullback_function(backend, _x -> conditions(_x, y, z; kwargs...), x)
+    pbA = pullback_function(backend, _y -> make_array(conditions(x, _y, z; kwargs...)), y)
+    pbB = pullback_function(backend, _x -> make_array(conditions(_x, y, z; kwargs...)), x)
     pbmA = PullbackMul!(pbA, size(y))
     pbmB = PullbackMul!(pbB, size(y))
-    Aᵀ_op = LinearOperator(R, m, m, false, false, pbmA)
+    Aᵀ_op = LinearOperator(R, m, m, false, false, pbmA)  # TODO: can it accept a vector if y is scalar?
```

Comment:
@baggepinnen the bug that made me give up was here, so if you find an elegant solution we'll be happy to implement it!